src/util/Stacktrace.cpp

```diff
@@ -90,21 +93,28 @@ void Stacktrace::printStracktrace(std::ostream& stream) {
     int trace_size = backtrace(trace.data(), trace.size());
     char** messages = backtrace_symbols(trace.data(), trace_size);
 
-    std::string exeName = getExePath();
-
     // skip first stack frame (points here)
-    for (int i = 1; i < trace_size; ++i) {
-        stream << "[bt] #" << i << " " << messages[i] << endl;
+    for (unsigned int i = 1; i < trace_size; ++i) {
+        stream << "[bt] #" << i - 1 << " " << messages[i] << endl;
+
+        Dl_info info;
+        dladdr(trace[i], &info);
 
         std::array<char, 1024> syscom{};
 
-        snprintf(syscom.data(), syscom.size(), "addr2line %p -e %s", trace[i], exeName.c_str());
+        snprintf(syscom.data(), syscom.size(), "addr2line %p -fCpe \"%s\"",
+                 static_cast<char*>(static_cast<char*>(nullptr) +
+                                    (static_cast<char*>(trace[i]) - static_cast<char*>(info.dli_fbase))),
```

Comment:
I think the "cleanest" solution would be to use something like that:
```
#include <cstdint>
#include <type_traits>
#include <cstring>

namespace xoj::util {
template<class To, class From>
inline
std::enable_if_t<
    sizeof(To) == sizeof(From) &&
    std::is_trivially_copyable_v<From> &&
    std::is_trivially_copyable_v<To>,
    To>
// constexpr support needs compiler magic
bit_cast(const From& src) noexcept {
    static_assert(std::is_trivially_constructible_v<To>,
        "This implementation additionally requires "
        "destination type to be trivially constructible");
    To dst;
    std::memcpy(&dst, &src, sizeof(To));
    return dst;
}

} 
```

and then you have to do the pointer arithmetic with uintptr_t like that:

```
auto offset = xoj::util::bit_cast<uintptr_t>(trace[i]) - xoj::util::bit_cast<uintptr_t>(info.dli_fbase) 
```

Btw, what's the `nullptr` to `char*` cast for? Is it because a nullptr is not necessarily int(0)?

Reply:
It was because `char*` - `char*` yields a `long int`, which I couldn't directly cast to a pointer type. And addition between `nullptr_type*` and `long int` isn't defined, while addition between `char*` and `long int` is.

Reply:
But I realized now after writing this that we don't need a pointer type as a result. We can just use the `%lx` format specifier. 
What would you say of 
```cpp
auto offset = reinterpret_cast<std::uintptr_t>(trace[i]) - reinterpret_cast<std::uintptr_t>(info.dli_fbase)
```

Reply:
I see. When you implement bit_cast, pls search if it has been implemented already. A good place to add it, is the safe_cast.h under the util/include directory.

Reply:
> But I realized now after writing this that we don't need a pointer type as a result. We can just use the `%lx` format specifier. 
> What would you say of this?
> ```cpp
> auto offset = reinterpret_cast<std::uintptr_t>(trace[i]) - reinterpret_cast<std::uintptr_t>(info.dli_fbase)
> ```

reinterpret casting to anything else than char* or the types pointer which originally was instantiated at that position s undefined behav. Pls use std::memcpy or std::bit_cast